external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py

```diff
@@ -287,11 +287,11 @@ def _init_recipe(self):
             logger.warning(f'train type {train_type} is not implemented yet.')
 
         self._recipe_cfg = MPAConfig.fromfile(recipe)
-        self._patch_datasets(self._recipe_cfg)  # for OTE compatibility
         self._patch_evaluation(self._recipe_cfg)  # for OTE compatibility
         logger.info(f'initialized recipe = {recipe}')
 
     def _init_model_cfg(self):
+        self._patch_datasets(self._recipe_cfg)  # for OTE compatibility, patch after overwriting hyperparams
```

Comment:
Is this 'init_model' method the right place for 'patch_dataset'?
@harimkang Could you recommend another place to put?

Reply:
Doing `patch_datasets` inside the `init_model_cfg` function doesn't seem appropriate. It seems to affect tasks that do not require datasets.

Reply:
I saw drop_true is determined by batch size in _patch_datasets. I think this code part should be inserted somewhere after loading init_hparam, not in patch_datasets.

Reply:
Reverted patch_datasets and moved to base_task.py. Please note cfg.drop_last will be parsed only in classification task, not for detection/segmentation task.